zebra-state/src/service.rs

```diff
@@ -679,3 +690,43 @@ impl Service<Request> for StateService {
 pub fn init(config: Config, network: Network) -> BoxService<Request, Response, BoxError> {
     BoxService::new(StateService::new(config, network))
 }
+
+/// Check if zebra is following a legacy chain and return an error if so.
+pub fn legacy_chain_check<I>(
```

Comment:
Small Module API: This function isn't called outside zebra-state
```suggestion
fn legacy_chain_check<I>(
```

Reply:
This suggestion was valid for when the function was in the same file as the prop tests. After the move of this function into service/arbitrary.rs we need it to be `pub` so we can call it from the prop tests. We now have it as `pub(crate)` as that is all we need.